python/pyspark/ml/classification.py

```diff
@@ -1560,8 +1581,9 @@ def trainSingleClass(index):
                             (classifier.predictionCol, predictionCol)])
             return classifier.fit(trainingDataset, paramMap)
 
-        # TODO: Parallel training for all classes.
-        models = [trainSingleClass(i) for i in range(numClasses)]
+        pool = ThreadPool(processes=min(self.getParallelism(), numClasses))
```

Comment:
So in Scala the threadpool is cached, here we aren't doing that and I think its a bit more heavy weight in Python so we might want to consider if there is a reasonable way to reuse (if not that's probably OK to since this overhead pales in comparison to training serially).